Dev/App.js

```diff
@@ -1,14 +1,42 @@
 import React, { Component } from 'react';
 import MainStackNavigator from './src/screens/MainStackNavigator';
 import { YellowBox } from 'react-native';
+import QCView from 'components/QCView';
+import screenStyle from 'config/screenStyle';
+import { checkNotifications, requestNotifications, check, request, PERMISSIONS } from 'react-native-permissions';
 
 export default class App extends Component {
 
+  state = {
+    requestingPermissions: true,
+  }
+
+  async componentDidMount() {
+
+    const isPermissionsGranted = await checkNotifications();
+    if (isPermissionsGranted !== 'granted') {
+      await requestNotifications(['alert', 'sound', 'badge']);
+      this.setState({ requestingPermissions: false });
+    } else {
+      this.setState({ requestingPermissions: false });
```

Comment:
If both the if clause and the else clause set the requestingPermissions to false, move it to after if(){}else(){}, so that you don't need to write it twice.